### ISSUE TITLE
Vue warn about getters not being side-effect free

### ISSUE THREAD

**Issue by DamianGlowala** [tailwindlabs/headlessui#2985]
**What package within Headless UI are you using?**

@headlessui/vue

**What version of that package are you using?**

insiders

**What browser are you using?**

Chrome

**Reproduction URL**

![image](https://github.com/tailwindlabs/headlessui/assets/48835293/bd41a7f8-a99c-4384-9860-e6ef3985d54c)
![image](https://github.com/tailwindlabs/headlessui/assets/48835293/74371191-7cf3-4889-ba2b-e20da75482e5)

https://vuejs.org/guide/essentials/computed.html#getters-should-be-side-effect-free

**Describe your issue**

Seeing Vue warnings about getters not being side-effect free for what to me seems like a Headless UI component.

**Comment by wahyubaskara** [comment]
Same issue

**Comment by faridibin** [comment]
Same Issue, started happening after last update

**Comment by SummerPaX** [comment]
I have also the same issue with headlessui tabs, and I did some digging.

Its possible that this computed might be one of the culprits, because it modifies the SSRContext.
```typescript
// packages/@headlessui-vue/src/components/tabs/tabs.ts

let mySSRIndex = computed(() => {
  if (SSRContext.value) {
    let mySSRIndex = SSRContext.value.panels.indexOf(props.id)
    if (mySSRIndex === -1) return SSRContext.value.panels.push(props.id) - 1
    return mySSRIndex
  }

  return -1
})
``` 

I don't think its that big of an issue because every computed is only executed twice and has no heavy code, but the warning messages are clogging up my console window.Thanks for doing such great wok

**Comment by m-shum** [comment]
Yup exactly the same issue here – happening for me specifically with the Tabs component of nuxtUI. The console noise is a headache for sure.

**Comment by MickL** [comment]
I think it helps more to just upvote the issue instead of writing "same issue" :)
